metaci/testresults/robot_importer.py

```diff
@@ -13,6 +15,20 @@ def import_robot_test_results(build_flow, path):
     # import is here to avoid import cycle
     from metaci.build.models import BuildFlowAsset
 
+    path = pathlib.Path(path)
+
+    # find the robot task that generated the log file so
+    # that we can access the task options when rendering
+    # the log file
+    mtime = timezone.make_aware(
+        datetime.fromtimestamp(path.stat().st_mtime), timezone.get_current_timezone()
+    )
+    robot_task = None
+    for task in build_flow.tasks.order_by("-time_end"):
+        if task.time_start <= mtime <= task.time_end:
```

Comment:
At first I thought doing this based on the timestamp felt brittle, but I guess it has the advantage of picking the right task if multiple tasks wrote to the same location.